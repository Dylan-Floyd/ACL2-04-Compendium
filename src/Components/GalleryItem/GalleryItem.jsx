import { namifyString } from '../../utils/utils.js'

import './GalleryItem.css'

export default function GalleryItem({ item }) {
  return (
    <div className='cardBorder'>
      <div className={`pokeItem cloudBG ${item.type_1}`}>
        <div className='topInfo'>
          <span className='name'>{namifyString(item.pokemon)}</span>
          <span className='hp'>{item.hp + 'HP'}</span>
        </div>
        <div className='imageBorder'>
          <div className='imageBG'>
            <img src={item.url_image} alt={item.pokemon} className='pokeImage' />
          </div>
        </div>
        <div className='bottomInfo'>
          {namifyString(item.type_1)} Pokemon. Length {item.height}, Weight: {item.weight} lbs.
        </div>
        <ul className='infoList'>
          <li>Type: {namifyString(item.type_1)}</li>
          <li>Speed: {item.speed}</li>
          <li>Ability 1: {namifyString(item.ability_1)}</li>
          <li>Base Pokemon: {namifyString(item.pokebase)}</li>
        </ul>
      </div>
    </div>
  )
}