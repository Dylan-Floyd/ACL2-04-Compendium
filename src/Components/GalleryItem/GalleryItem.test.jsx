import { render } from '@testing-library/react'
import GalleryItem from './GalleryItem.jsx'

it('Should render a pokemon', () => {
  const { container } = render(<GalleryItem item={
    {
      type_1: 'grass',
      pokemon: 'bob',
      hp: 3,
      url_image: 'wow',
      height: 3,
      weight: 3,
      speed: 3,
      ability_1: 'wilson wow',
      pokebase: 'asdf'
    }
  } />)

  expect(container).toMatchSnapshot()
})