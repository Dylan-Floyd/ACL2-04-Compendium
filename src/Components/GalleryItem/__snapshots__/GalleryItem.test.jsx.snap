// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a pokemon 1`] = `
<div>
  <div
    class="cardBorder"
  >
    <div
      class="pokeItem cloudBG grass"
    >
      <div
        class="topInfo"
      >
        <span
          class="name"
        >
          Bob 
        </span>
        <span
          class="hp"
        >
          3HP
        </span>
      </div>
      <div
        class="imageBorder"
      >
        <div
          class="imageBG"
        >
          <img
            alt="bob"
            class="pokeImage"
            src="wow"
          />
        </div>
      </div>
      <div
        class="bottomInfo"
      >
        Grass 
         Pokemon. Length 
        3
        , Weight: 
        3
         lbs.
      </div>
      <ul
        class="infoList"
      >
        <li>
          Type: 
          Grass 
        </li>
        <li>
          Speed: 
          3
        </li>
        <li>
          Ability 1: 
          Wilson Wow 
        </li>
        <li>
          Base Pokemon: 
          Asdf 
        </li>
      </ul>
    </div>
  </div>
</div>
`;
